### PR TITLE
Merge PR

### PR DESCRIPTION
**Pull request by Jaaw1d**

### CODE REVIEW

**Comment by Jaaw1d** [comment]
Good luck

File: src/mergeSort.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    // Sort two halves of split array
+    const leftSortedArray = this.sort(leftArray);
+    const rightSortedArray = this.sort(rightArray);
+
+    // Merge two sorted arrays into one.
+    return this.mergeSortedArrays(leftSortedArray, rightSortedArray);
+  }
+
+  mergeSortedArrays(leftArray, rightArray) {
+    const sortedArray = [];
```
Comment (Cavidiko) [inline src/mergeSort.js:27]: Haha

**Review by Caqqolu — CHANGES_REQUESTED** [review]

**Comment by Jaaw1d** [comment]
Admin here, please review more carefully this time!

File: src/countingSort [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    for (let bucketIndex = 1; bucketIndex < buckets.length; bucketIndex += 1) {
+      buckets[bucketIndex] += buckets[bucketIndex - 1];
+    }
+
+    // Now let's shift frequencies to the right so that they show correct numbers.
+    // I.e. if we won't shift right than the value of buckets[5] will display how many
+    // elements less than 5 should be placed to the left of 5 in sorted array
+    // INCLUDING 5th. After shifting though this number will not include 5th anymore.
+    buckets.pop();
+    buckets.unshift(0);
```
Comment (Cavidiko) [inline src/countingSort:54]: Wow

**Review by Caqqolu — APPROVED** [review]

**Comment by Jaaw1d** [comment]
Good job, merging the PR

**Pull request merged**